### PR TITLE
[server-wallet] Clean store.ts

### PR DESCRIPTION
**Pull request by snario**
Ran through some (not all) functions in the `Store` implementation and tried to simplify them using generally reasonable coding standards.

### CODE REVIEW

File: packages/server-wallet/src/wallet/store.ts [hunk 1]
```diff
@@ -455,101 +444,79 @@ export class Store {
...
+   * @param channelId - the channel ID the state refers to
+   * @param state - the new signed state to be added
+   * @param tx - an existing database transaction context
+   *
+   * This function serves as both a data validator (ensure the state is valid) and
+   * an efficient operator on the database.
+   */
   async addSignedState(
     channelId: string,
-    maybeChannel: Channel | undefined,
```
Comment (geoknee) [inline packages/server-wallet/src/wallet/store.ts:460]: By ripping out this parameter, you are making this function go to the DB _for sure_. Previously, I could skip that if  I already had the channel?

File: packages/server-wallet/src/wallet/store.ts [hunk 2]
```diff
@@ -455,101 +444,79 @@ export class Store {
...
-    wireSignedState: WireSignedState,
+    wireState: WireSignedState,
```
Comment (geoknee) [inline packages/server-wallet/src/wallet/store.ts:459]: why?

File: packages/server-wallet/src/wallet/store.ts [hunk 3]
```diff
@@ -632,103 +599,75 @@ class StoreError extends WalletError {
...
-    {
-      ...constants,
-      channelId,
-      vars: [],
-      chainServiceRequests: [],
-      signingAddress,
-      fundingStrategy,
-    },
-    ...CHANNEL_COLUMNS
+  await Nonce.ensureLatest(constants.channelNonce, addresses, txOrKnex);
```
Comment (geoknee) [inline packages/server-wallet/src/wallet/store.ts:620]: This is _new_ behaviour, if I am not mistaken, and not just a refactor?

File: packages/server-wallet/src/wallet/__test__/integration/push-message.test.ts [hunk 4]
```diff
@@ -67,7 +67,7 @@ it('stores states contained in the message, in a single channel model', async ()
   expect(channelsAfter[0].vars).toHaveLength(2);
 
   // The Channel model adds the state hash before persisting
-  expect(signedStates.map(addHash)).toMatchObject(channelsAfter[0].vars);
+  expect(channelsAfter[0].vars).toMatchObject(signedStates);
```
Comment (geoknee) [inline packages/server-wallet/src/wallet/__test__/integration/push-message.test.ts:70]: Is the addHash not necessary?

**Review by geoknee — APPROVED** [review]

File: packages/server-wallet/src/wallet/store.ts (hunk 1 above)
Comment (snario) [reply packages/server-wallet/src/wallet/store.ts:460]: It was never used in any instance of the codebase, so I removed it for that reason

File: packages/server-wallet/src/wallet/store.ts (hunk 2 above)
Comment (snario) [reply packages/server-wallet/src/wallet/store.ts:459]: Just shortening the code — not attached to this

File: packages/server-wallet/src/wallet/store.ts (hunk 3 above)
Comment (snario) [reply packages/server-wallet/src/wallet/store.ts:620]: Same behaviour as before! It used to be a bit more hidden

File: packages/server-wallet/src/wallet/__test__/integration/push-message.test.ts (hunk 4 above)
Comment (snario) [reply packages/server-wallet/src/wallet/__test__/integration/push-message.test.ts:70]: No, it was being used to mask an internal implementation detail

**Pull request merged**